File: FunikiSDKExample/FunikiSDKExample/FunikiDevicePickerViewController.m
//
//  Created by Matilde Inc.
//  Copyright (c) 2015 Fun'iki Project. All rights reserved.
//

#import "FunikiDevicePickerViewController.h"
#import "MAFunikiManager.h"

@interface FunikiDevicePickerViewController () <MAFunikiManagerDelegate>

@end

@implementation FunikiDevicePickerViewController

#pragma mark - UIViewController
- (void)viewDidLoad
{
    [super viewDidLoad];
    self.title = @"雰囲気メガネを選択";
}

- (void)viewWillAppear:(BOOL)animated
{
    [[MAFunikiManager sharedInstance] setDelegate:self];
    [[MAFunikiManager sharedInstance] startSelectingDevice];
    
    [self.tableView reloadData];
    
    [super viewWillAppear:animated];
}

- (BOOL)shouldAutorotate
{
    return NO;
}

#pragma mark - MAFunikiManagerDelegate
- (void)funikiManager:(MAFunikiManager *)sdk didUpdateDiscoveredPeripherals:(NSArray *)peripherals
{
    [self.tableView reloadData];
}

#pragma mark - UITableViewDelegate/DataSource
- (NSInteger)numberOfSectionsInTableView:(UITableView *)aTableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)aTableView numberOfRowsInSection:(NSInteger)section
{
    return [[[MAFunikiManager sharedInstance] discoveredPeripherals] count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"CellIdentifier" forIndexPath:indexPath];
    
    MADiscoveredPeripheral *discoveredPeripheral = [[[MAFunikiManager sharedInstance] discoveredPeripherals] objectAtIndex:indexPath.row];
    cell.textLabel.text = discoveredPeripheral.peripheral.name;

    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSArray *peripherals = [[MAFunikiManager sharedInstance] discoveredPeripherals];
    MADiscoveredPeripheral *selectedPeripheral = peripherals[indexPath.row];
    if (selectedPeripheral){
        [[MAFunikiManager sharedInstance] connectPeripheral:selectedPeripheral];
    }
    [[MAFunikiManager sharedInstance] stopSelectingDevice];
    [self dismissViewControllerAnimated:YES completion:nil];
}

#pragma mark - Action
- (IBAction)cancel:(id)sender
{
    [[MAFunikiManager sharedInstance] stopSelectingDevice];
    [[self presentingViewController] dismissViewControllerAnimated:YES completion:NULL];
}

@end

File: FunikiSDKExampleSwift/FunikiSDKExampleSwift/FunikiDevicePickerViewController.swift
//
//  Created by Matilde Inc.
//  Copyright (c) 2015 FUN'IKI Project. All rights reserved.
//

import UIKit

class FunikiDevicePickerViewController: UITableViewController, MAFunikiManagerDelegate {
    
    let funikiManager = MAFunikiManager.sharedInstance()
    
    // MARK: - UITableViewController
    override func viewDidLoad(){
        super.viewDidLoad()
        self.title = "雰囲気メガネを選択"
    }
    
    override func viewWillAppear(_ animated: Bool) {
        funikiManager?.delegate = self
        funikiManager?.startSelectingDevice()
        
        self.tableView.reloadData()
        
        super.viewWillAppear(animated)
    }
    
    override var shouldAutorotate: Bool{
        return false
    }
    
    
    
    // MARK: - MAFunikiManagerDelegate
    private func funikiManager(manager: MAFunikiManager!, didUpdateDiscoveredPeripherals peripherals: [AnyObject]!) {
        self.tableView.reloadData()
    }
    
    // MARK: - UITableViewDataSource/Delegate
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        let count = funikiManager!.discoveredPeripherals.count
        return count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "CellIdentifier", for: indexPath) as UITableViewCell
        
        let discoveredPeripheral = funikiManager?.discoveredPeripherals[indexPath.row] as! MADiscoveredPeripheral
        cell.textLabel?.text = discoveredPeripheral.peripheral.name
        
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let discoveredPeripheral = funikiManager?.discoveredPeripherals[indexPath.row] as! MADiscoveredPeripheral
        funikiManager?.connect(discoveredPeripheral)
        funikiManager?.stopSelectingDevice()
        self.dismiss(animated: true, completion:nil)
    }
    
    // MARK: - Action
    @IBAction func cancel(sender:AnyObject!) {
        funikiManager?.stopSelectingDevice()
        self.dismiss(animated: true, completion: nil)
    }
}
